cmd/argocd/commands/app.go

```diff
@@ -759,7 +733,7 @@ func printAppResources(w io.Writer, app *argoappv1.Application, showOperation bo
 }
 
 // printAppStateChange prints a component state change if it was different from the last time we saw it
-func printAppStateChange(w io.Writer, prevComp *argoappv1.ComparisonResult, app *argoappv1.Application) {
+func printAppStateChange(w io.Writer, prevComp *argoappv1.ComparisonResult, app *argoappv1.Application, forSync, forRollback bool) {
```

Comment:
The method signature of this function doesn't have enough information of printing state changes for an application operation. Meaning, it only accepts `prevComp * argoappv1.ComparisonResult ` as an input, but `argoappv1.ComparisonResult ` does not know the previous hook statuses, which is captured only in `Status.OperationState`.  I don't believe this will work.

For printing a state change which includes operations, you will need the previous app.Status.OperationState